cli/internal/kubernetes/upgrade.go

```diff
@@ -343,6 +347,71 @@ func (u *Upgrader) GetClusterAttestationConfig(ctx context.Context, variant vari
 	return existingAttestationConfig, existingConf, nil
 }
 
+// ExtendClusterConfigCertSANs extends the ClusterConfig stored under "kube-system/kubeadm-config" with the given SANs.
+// Existing SANs are preserved.
+func (u *Upgrader) ExtendClusterConfigCertSANs(ctx context.Context, alternativeNames []string) error {
```

Comment:
Could separate getting/updating of cluster resources from data processing for easier unit testing.
Not sure if this would need a unit test. I am leaning towards "no".

Reply:
I think the current implementation makes it easier to refactor this to use retryOnConflict later.
The retry loop always needs to get, patch locally, update.
So I'd leave this as-is for now.